processing/src/main/java/org/apache/druid/storage/local/LocalTmpStorage.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.storage.local;
+
+import com.google.inject.Provider;
+import org.apache.druid.java.util.common.FileUtils;
+
+import java.io.File;
+
+public interface LocalTmpStorage
+{
+  /**
+   * Get a temporary directory.
+   *
+   * @return a temporary directory
+  */
+  File getTmpDir();
+
+  class DefaultLocalTmpStorageProvider implements Provider<LocalTmpStorage>
```

Comment:
doesn't seem like the system will prepare to clean up these files - wouldn't that will fill up the disk/create garabge?

Reply:
The tmpdir provided from this is just a location.  Whether things are cleaned up or not is a question of the implementation.  In general, most of the code that deals with tmp files cleans them up after itself and if it's not cleaning up after itself, that should either be intentional and have a design reason or it is a bug in the code that's dealing with the file, not a bug in the code that's delivering the location of the directory.